lib/req/test.ex

```diff
@@ -167,6 +167,12 @@ defmodule Req.Test do
   @typedoc since: "0.4.15"
   @opaque stub() :: atom()
 
+  if Code.ensure_loaded?(Plug.Conn) do
+    @type plug() :: {module(), [term()]} | module() | (Plug.Conn.t() -> Plug.Conn.t())
+  else
+    @type plug() :: {module(), [term()]} | module() | (term() -> term())
```

Comment:
strictly speaking fun/2 is part of plug contract but fun/1 allegedly is _not_ (https://github.com/elixir-plug/plug/issues/1201#issuecomment-1872276119) so we should include it:

```suggestion
    @type plug() :: {module(), [term()]} | module() | (Plug.Conn.t() -> Plug.Conn.t() | (Plug.Conn.t(), term() -> Plug.Conn.t())
  else
    @type plug() :: {module(), [term()]} | module() | (term() -> term() | (term(), term() -> term())
```

Reply:
Oh very good catch!!